File: component.py
"""
Components

Components are a shell parent class that have metadata fields.

- Component Name
- Component Description
"""

class Component:
    def __init__(self,name,description):
        self.name = name 
        self.description = description

    def toString(self):
        print("{}: {}".format(self.name,self.description))
   
def main(): 
    testComponent = Component("test","test component description")
    testComponent.toString()
   
# __name__ 
if __name__=="__main__": 
    main()

File: kotlin_prototype/tether_android_kotlin/app/src/main/java/com/example/mykotlinapp/Component.kt
package com.example.mykotlinapp

/*
Components

Components are a shell parent class that have metadata fields.

- Component Name
- Component Description
 */

abstract class Component(_name: String, _description: String){
    var name: String
    var description: String

    init {
        this.name = _name
        this.description = _description
    }

    // Override the toString() method to return a string of relevant data
    override fun toString(): String {
        var string: String = "Name: " + this.name + " | Description: " + this.description
        return string
    }

    abstract fun action():String

}